src/vos/evtree.c

```diff
@@ -1178,12 +1184,16 @@ evt_tcx_create(struct evt_root *root, uint64_t feats, unsigned int order,
 
 		tcx->tc_feats	= root->tr_feats;
 		tcx->tc_order	= root->tr_order;
+		if (tcx->tc_feats & EVT_FEAT_DYNAMIC_ROOT)
+			tcx->tc_max_order = root->tr_max_order;
+		else
+			tcx->tc_max_order = root->tr_order;
```

Comment:
I'm not sure I understand the question.

If the feature is enabled, tr_order will be the current order and tr_max_order will be the user-specified tree order.  If the feature is not enable, only tr_order will be set and it will always be the user-specified tree order.  tcx->tc_max_order always gets set to the user specified tree order whereas tcx->tc_order will be the current dynamic tree order (or the user-specified value when the feature is disabled).